Detectors/MUON/MCH/DigitFiltering/src/DigitModifier.cxx

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "MCHDigitFiltering/DigitModifier.h"
+
+#include "DataFormatsMCH/Digit.h"
+#include "MCHMappingInterface/Segmentation.h"
+#include <functional>
+#include <array>
+#include <unordered_map>
+#include <format>
+#include <stdexcept>
+
+namespace
+{
+
+o2::mch::DigitModifier createST1MappingCorrector(int /*runNumber*/)
+{
+  return {};
+}
+
+/** initialization of the pad remapping table for Station 2 DEs
+ */
+void initST2PadsRemappingTable(std::unordered_map<int, std::unordered_map<int, int>>& padsRemapping)
+{
+  // Remapping of ST2 DS boards near the rounded part
+  {
+    std::array<int, 8> deToRemap{300, 301, 302, 303, 400, 401, 402, 403};
+    std::array<int, 5> dsToRemap{99, 100, 101, 102, 103};
+
+    for (auto deId : deToRemap) {
+
+      const o2::mch::mapping::Segmentation& segment = o2::mch::mapping::segmentation(deId);
+      for (auto dsId : dsToRemap) {
+        // double loop on DS channels
+        // 1. find the minimum pad index of the DS board
+        int padIdMin = -1;
+        int channelForPadIdMin = -1;
+        for (int channel = 0; channel < 64; channel++) {
+          auto padId = segment.findPadByFEE(dsId, int(channel));
+          if (padId < 0) {
+            // this should never occur in this specific case, as all channels of this group of boards
+            // is connected to pads, hence we rise an exception
+            throw std::out_of_range(std::format("Unknown padId for DE{} DS{} channel {}", deId, dsId, channel));
+          }
```

Comment:
Hi @aferrero2707 , the test on MacOS fails because of std::format. I think C++20 is not yet fully supported and we should still use fmt::format.

Reply:
Thanks, replaced with `fmt::format()`.